lib/Sema/PreCheckExpr.cpp

```diff
@@ -1056,6 +1056,29 @@ namespace {
           if (isa<SequenceExpr>(parent))
             return finish(true, expr);
 
+          // Unwrap to the outermost paren in the sequence.
+          if (isa<ParenExpr>(parent)) {
+            for (;;) {
+              auto nextParent = parents.find(parent);
+              if (nextParent == parents.end())
+                break;
+
+              // e.g. `foo((&bar), x: ...)`
+              if (isa<TupleExpr>(nextParent->second)) {
```

Comment:
I’m not sure this belongs inside the loop. We don’t allow & inside tuple literals, only at the top level of an argument list.

Reply:
The actual checking is done a bit later, it would check whether paren or tuple represent an argument to a call or subscript, this loop just walk up the parens until it finds the outmost paren or a tuple.